src/account/sync/mod.rs

```diff
@@ -736,6 +738,12 @@ async fn perform_transfer(
 ) -> crate::Result<Message> {
     let mut utxos = vec![];
     let mut transaction_inputs = vec![];
+    // store (amount, address, new_created) to check later if dust is allowed
+    let mut dust_and_allowance_recorders = Vec::new();
+
+    if transfer_obj.amount.get() < DUST_ALLOWANCE_VALUE {
```

Comment:
Needs to be adapted at a later point, if you want to create a dust allowance output and dust in a single transaction (if you ever want to do that)